src/cfnlint/maintenance.py

```diff
@@ -186,3 +243,171 @@ def update_iam_policies():
 
     with open(filename, 'w') as f:
         json.dump(content, f, indent=2, sort_keys=True, separators=(',', ': '))
+
+
+def get_schema_value_types():
+
+    def resolve_refs(properties, schema):
+        results = {}
+        name = None
+
+        if properties.get('$ref'):
+            name = properties.get('$ref').split('/')[-1]
+            subname, results = resolve_refs(schema.get('definitions').get(name), schema)
+            if subname:
+                name = subname
+            properties = schema.get('definitions').get(name)
+
+        if properties.get('type') == 'array':
+            results = properties.get('items')
+
+        if results:
+            properties = results
+
+        if results and results.get('$ref'):
+            name, results = resolve_refs(results, schema)
+
+        if not results:
+            return name, properties
+
+        return name, results
+
+    def get_object_details(names, properties, schema):
+        results = {}
+        for propname, propdetails in properties.items():
+            subname, propdetails = resolve_refs(propdetails, schema)
+            t = propdetails.get('type')
+            if not t:
+                continue
+            if t in ['object']:
+                if subname is None:
+                    subname = propname
+                if propdetails.get('properties'):
+                    if subname not in names:
+                        results.update(get_object_details(
+                            names + [subname], propdetails.get('properties'), schema))
+                elif propdetails.get('oneOf') or propdetails.get('anyOf') or propdetails.get('allOf'):
+                    LOGGER.info(
+                        'Type %s object for %s has only oneOf,anyOf, or allOf properties', names[0], propname)
+                    continue
+            elif t not in ['string', 'integer', 'number', 'boolean']:
+                if propdetails.get('$ref'):
+                    results.update(get_object_details(
+                        names + [propname], schema.get('definitions').get(t.get('$ref').split('/')[-1]), schema))
+                elif isinstance(t, list):
+                    LOGGER.info('Type for %s object and %s property is a list', names[0], propname)
+                else:
+                    LOGGER.info('Unable to handle %s object for %s property', names[0], propname)
+            elif t == 'string':
+                if not results.get('.'.join(names + [propname])):
+                    if propdetails.get('pattern') or (propdetails.get('minLength') and propdetails.get('maxLength')) or propdetails.get('enum'):
+                        results['.'.join(names + [propname])] = {}
+                if propdetails.get('pattern'):
+                    p = propdetails.get('pattern')
+                    if '.'.join(names + [propname]) == 'AWS::OpsWorksCM::Server.CustomPrivateKey':
```

Comment:
[linking for reference](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-opsworkscm/blob/a9c53dda37360c4b6497fa9283ee9a8922687ef1/aws-opsworkscm-server/aws-opsworkscm-server.json#L82)